ouroboros-network/src/Ouroboros/Network/PeerSelection/Governor/Monitor.hs

```diff
@@ -113,7 +120,17 @@ connections PeerSelectionActions{peerStateActions = PeerStateActions {monitorPee
                           -- handled elsewhere. We just update the async ones:
                           establishedStatus = demotedToWarm
                                                 <> establishedStatus
-                                                Map.\\ demotedToCold
+                                                Map.\\ demotedToCold,
+
+                          -- Asynchronous transition to cold peer can only be
+                          -- a result of a failure.
+                          knownPeers        = KnownPeers.setConnectTime
+                                                (Map.keysSet demotedToCold)
+                                                (reconnectDelay `addTime` now)
+                                            . foldr
+                                                ((snd .) . KnownPeers.incrementFailCount)
```

Comment:
We're folding a set. There's not a big difference between right and left fold (at least as comparing to, e.g. a list): [foldr](https://hackage.haskell.org/package/containers-0.6.4.1/docs/src/Data.Set.Internal.html#foldr).  But we could use the strict version, to force evaluation of the known peers map at each step.

Reply:
> This will be easier if we don't have incrementFailCount returning a second result.

But we also need the first result [else where](https://github.com/input-output-hk/ouroboros-network/pull/2526/files#diff-bf40eeb6032710c0df55849a2014a47fb5102878fa5df042ff6672a3ed688c36R135).